src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateEntry.cs

```diff
@@ -1,12 +1,14 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Collections.Generic;
+
 namespace Microsoft.AspNetCore.Mvc.ModelBinding
 {
     /// <summary>
     /// An entry in a <see cref="ModelStateDictionary"/>.
     /// </summary>
-    public class ModelStateEntry
+    public abstract class ModelStateEntry
```

Comment:
Previously users were able to create a `ModelStateEntry` and `Add` it to the dictionary. Not sure why this is no longer the case but the inconsistency between making `ModelStateEntry` harder to create and `ModelStateDictionary` having so much public surface for adding / inserting `ModelStateEntry`s isn't good. Suggest returning this to non-`abstract`. (If we really don't want users to create instances, must also `throw` a `NotSupportedException` in the relevant `ModelStateDictionary` methods. But, again, why?)